express/parsers/structure.py

```diff
@@ -41,6 +68,86 @@ def __init__(self, *args, **kwargs):
         self.lattice_only_structure = mg.Structure.from_str(self.structure_string, self.structure_format)  # deepcopy
         self.lattice_only_structure.remove_sites(range(1, len(self.structure.sites)))
 
+    def get_inchi_run(self):
+        return inchi_run
+
+    def get_inchi_null(self):
+        """
+        Function returns "Not Available" value for InChI when it cannot be calculated.
+
+        Returns:
+            Str
+        """
+        inchi_key_str = {
+            'name': 'inchi',
+            'value': ''
+        }
+        return inchi_key_str
+
+    def create_pybel_smi_from_poscar(self):
+        """
+        Function using ase to convert the POSCAR formatted string of a structure
+        into an XYZ formatted text file for that structure.
+
+        Then pybel converts the XYZ formatted text file into a SMILES format.
+
+        Returns:
+            Str
+        """
+        fi = "geom.xyz"
+        f = open(fi, "w")
+        os.chmod(fi, 0o777)
+        fstr = StringIO(self.structure_string)
+        ase_pos = ase.io.read(fstr, format="vasp")
+        xyz_file = ase.io.write(fi, ase_pos, format='xyz')
+        psmi = list(pybel.readfile('xyz', 'geom.xyz'))[0]
+        f.close()
+        return psmi
+
+    def get_inchi(self):
+        """
+        Function calculates the International Chemical Identifier (InChI) string for a given structure.
+
+        Returns:
+            Str
+        """
+        psmi = self.create_pybel_smi_from_poscar()
+        self.inchi = psmi.write("inchi")
+        inchi_short = self.inchi.split("=")
+        inchi_short = inchi_short[1]
+        inchi_str = {
+            "name": "inchi",
+            "value": inchi_short
+        }
+        return inchi_str
+
+    def get_inchi_key_null(self):
+        """
+        Function returns "Not Available" for the non-human readable InChI Hash value when it is unavailable
+
+        Returns:
+            Str
+        """
+        inchi_key_str = {
+            'name': 'inchi_key',
+            'value': ''
+        }
+        return inchi_key_str
+
+    def get_inchi_key(self):
+        """
+        Function calculates the non-human readable InChI Hash value.
+
+        Returns:
+            Str
+        """
+        inchi_key_val = rdkit.Chem.inchi.InchiToInchiKey(self.inchi)
+        inchi_key_str = {
+            "name": "inchi_key",
+            "value": inchi_key_val
+        }
```

Comment:
Lines 145-148, 131-134 and 118-121 are practically identical, we should reuse the code